### PR TITLE
Epic annotation UI

### PR DESCRIPTION
**Pull request by INRokh**
1. Allow admin to assign user to annotation
2. Show list of annotations to user 
3. User can choose image for annotation

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -73,8 +74,9 @@ class App extends Component {
           <Link to="/">Annotations</Link> | <Link to="/logout">Logout</Link>
           <Switch>
             <Route exact path="/">
-              <ImageAnnotation />
+              <AnnotationList />
             </Route>
+            <Route path = "/annotations/:id" children={<AnnotationPage />} />
```
Comment (INRokh) [inline src/App.js:79]: use [route url params](https://reacttraining.com/react-router/web/example/url-params ) to show annotation by id

File: src/actions/index.js [hunk 2]
```diff
@@ -32,13 +25,32 @@ export const fetchImages = () => {
...
-// export const createImages = image => {
-//   return async (dispatch, getState) => {
-//     const response = await LocalApi.post("/images", image);
-//     return dispatch({
-//       type: "SET_IMAGES",
-//       payload: response.data
-//     });
-//   };
-// };
+export const fetchUsers = () => {
```
Comment (INRokh) [inline src/actions/index.js:28]: wrappers around api calls

File: src/components/AnnotationList.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React, {Component} from 'react';
```
Comment (INRokh) [inline src/components/AnnotationList.js:1]: This is a component that shows the annotations assigned to the current user

File: src/components/images_uploading/DisplayImageList.js [hunk 4]
```diff
@@ -1,36 +1,63 @@
-import React, { Component } from "react";
+import React, { Component, useReducer } from "react";
 import { connect } from "react-redux";
-import { fetchImages } from "./../../actions";
-import { Button } from "react-bootstrap";
+import { fetchImages, fetchUsers } from "./../../actions";
+import LocalApi from "../../apis/LocalApi";
+
+
+class UserAssignmentDropDown extends React.Component {
```
Comment (INRokh) [inline src/components/images_uploading/DisplayImageList.js:7]: it's temporary here, need to move to another file. This component shows a dropdown with list of users next to an image. When a user is selected in the dropdown an Assign button is shown.  
[tutorial](https://reactjs.org/docs/forms.html#the-select-tag)

File: src/components/images_uploading/DisplayImageList.js [hunk 5]
```diff
@@ -1,36 +1,63 @@
...
-                className="m-2"
-                href={`${process.env.REACT_APP_BASEURL}/images/${image.s3key}/annotate`}
-              >
-                Add Annotation
-              </Button>
-            </li>
+              <li key={image._id}>
+                {image.lot} {image.unitNumber} {image.productDescription}
+                <UserAssignmentDropDown imageId={image._id} users={users} />
+                <a
```
Comment (INRokh) [inline src/components/images_uploading/DisplayImageList.js:55]: buttons were broken for me, so I temporarily replaced it with <a>

File: src/components/images_uploading/DisplayImageList.js [hunk 6]
```diff
@@ -40,8 +67,9 @@ class DisplayImageList extends Component {
 
 const mapStateToProps = state => {
   return {
-    images: state.images
+    images: state.images,
+    users: state.admin.users
```
Comment (INRokh) [inline src/components/images_uploading/DisplayImageList.js:71]: added list of all users

**Review by EllieChen-Git — APPROVED** [review]
Wow, Nina, this is great!!!!! You are amazing!!! ❤️❤️❤️

Sorry I just shut down my computer (so I won’t be able to test the code or do the merge), but I read through all the changes you made on my mobile (I’m actually on bed now 🛌 😆) and they look quite promising to me!!!

Thanks heaps for your hard work! I’m going to sleep now. You should take a good rest too!
I’ll talk to you tmr. Good night 🌙

—-
Ohoh, sorry I just noticed that you are still adding comments! I’ll have a look tmr, but most importantly you should go to bed now too 😆

File: src/components/pages/AnnotationPage.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import React, {Component}  from "react";
+import { connect } from "react-redux";
+import { Link, useParams } from "react-router-dom";
+import { fetchAnnotationDetails } from "../../actions";
+import ImageAnnotation from "../ImageAnnotation";
+
+function AnnotationPage (props) {
+  let { id } = useParams();
+  const { annotation } = props;
+  if (!annotation || annotation._id != id) {
```
Comment (INRokh) [inline src/components/pages/AnnotationPage.js:10]: if annotation is empty we need to load it. Or if annotation id is different to annotation id from the url this means that we need to reload it.

File: src/components/pages/AnnotationPage.js [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import React, {Component}  from "react";
+import { connect } from "react-redux";
+import { Link, useParams } from "react-router-dom";
+import { fetchAnnotationDetails } from "../../actions";
+import ImageAnnotation from "../ImageAnnotation";
+
+function AnnotationPage (props) {
```
Comment (INRokh) [inline src/components/pages/AnnotationPage.js:7]: Added annotation page that wraps image annotation providing image src url to the image annotation component

File: src/reducers/auth_reducer.js [hunk 9]
```diff
@@ -1,4 +1,5 @@
 const defaultState = {
+  user: null,
```
Comment (INRokh) [inline src/reducers/auth_reducer.js:2]: added default user value

**Pull request merged**